lib/fizzy/instructions.cpp

```diff
@@ -0,0 +1,231 @@
+#include "instructions.hpp"
+
+namespace fizzy
+{
+namespace
+{
+constexpr auto variadic{0};
+
+constexpr InstructionMetrics instruction_metrics_table[256] = {
+    // 5.4.1 Control instructions
+    /* unreachable         = 0x00 */ {0, 0},
+    /* nop                 = 0x01 */ {0, 0},
+    /* block               = 0x02 */ {0, 0},
+    /* loop                = 0x03 */ {0, 0},
+    /* if_                 = 0x04 */ {1, -1},
+    /* else_               = 0x05 */ {0, 0},
+    /*                       0x06 */ {},
+    /*                       0x07 */ {},
+    /*                       0x08 */ {},
+    /*                       0x09 */ {},
+    /*                       0x0a */ {},
+    /* end                 = 0x0b */ {0, 0},
+    /* br                  = 0x0c */ {variadic, 0},
+    /* br_if               = 0x0d */ {variadic, 0},
```

Comment:
`br_if` an `br_table` pop one value from the stack

Reply:
You are absolutely right. I confused myself that this depends on the target label. This should reduce complexity of this work by 10. Thanks.